src/components/animated-button/AnimatedButtonProvider.tsx

```diff
@@ -47,7 +47,10 @@ const AnimatedButton = ({
     if (!isPlaying) {
       setIconIndex(i)
     }
-  }, [i, setIconIndex, isPlaying])
+    // Intentionally omitting `isPlaying` from the dependencies,
+    // this effect should only trigger when `i` is updated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [i, setIconIndex])
```

Comment:
This seems like a potential future foot-gun if not a current bug:
https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528

Maybe use a ref here to track previous `i` values and compare?

Reply:
Ok it's a little more verbose, but this works just as well. What do you think? https://github.com/AudiusProject/audius-mobile-client/pull/249/commits/093be697c056edf8d72a6af71cf9c0c74af87f8e#diff-c9dd9b94e8aadccfa0902fa7e196648983db14a833e54b5c9e786f3c8c32680c